src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,194 @@
+import { Component } from 'react';
+import { div, form, input, label, textarea, hh, select, option } from 'react-hyperscript-helpers';
+import { SupportRequestBaseModal } from '../SupportRequestBaseModal';
+import { Alert } from '../Alert';
+import { Support, User } from '../../libs/ajax';
+import { Storage } from "../../libs/storage";
+import { convertCompilerOptionsFromJson } from 'typescript';
+
+export const SupportRequestModal = hh(class SupportRequestModal extends Component {
+  constructor(props) {
+    super(props);
+    const isLogged = Storage.userIsLogged();
+    let name = isLogged ? Storage.getCurrentUser().displayName : '';
+    let first_name = isLogged ? ", " + name.split(" ")[0] : '';
+    let email = isLogged ? Storage.getCurrentUser().email : '';
+    let height = isLogged ? "550px": '700px';
+    let top = isLogged ? '400px': '1';
+    this.state = {
+      name: name,
+      isLogged: isLogged,
+      type: 'question',
+      subject: '',
+      description: '',
+      attachment: '',
+      email: email,
+      first_name: first_name,
+      height: height,
+      top: top,
+      vaild: false
+    };
+    this.closeHandler = this.closeHandler.bind(this);
+    this.afterOpenHandler = this.afterOpenHandler.bind(this);
+    this.OKHandler = this.OKHandler.bind(this);
+    this.nameChangeHandler = this.nameChangeHandler.bind(this);
+    this.typeChangeHandler = this.typeChangeHandler.bind(this);
+    this.subjectChangeHandler = this.subjectChangeHandler.bind(this);
+    this.descriptionChangeHandler = this.descriptionChangeHandler.bind(this);
+    this.attachmentChangeHandler = this.attachmentChangeHandler.bind(this);
+    this.emailChangeHandler = this.emailChangeHandler.bind(this);
```

Comment:
You can avoid all of these boilerplate bindings by using the arrow syntax, i.e.
```
functionName = () => {
  // do something
};
```
Or for async functions ... 
```
functionName = async () => {
  // do something
};
```
If they are plain functions, then you have to use these binding statements in the constructor, which I find annoying and error prone.